### PR TITLE
fixed controls

### LINKED ISSUE #207: compile inputs into a dictionary
## Task Description

- compile inputs into a dictionary (judge_id=None, circuit_id=None, gender=None, appointment_date=None, judge_type_id=None)
- turn circuit and judge type into corresponding IDs

## User Stories

- As data engineers of this project, we need to organise our code in a way such that as we work on different parts of the dashboard, we still have consistent variable names.

## Relevant Files [If Available] 

-

### PR DESCRIPTION
**Pull request by RachelPun**
closes #207

### CODE REVIEW

File: dashboard/streamlit_app.py [hunk 1]
```diff
@@ -34,7 +34,11 @@ def get_db_connection(config) -> connect:
 
 def extract_id_from_string(string: str) -> int:
     """Returns id in a bracket before string info."""
-    return int(re.match(r"\((\d+)\)", string).group(1))
+
+    if string:
+        return int(re.match(r"\((\d+)\)", string).group(1))
```
Comment (ErvinRex) [inline dashboard/streamlit_app.py:39]: Great regex to catch ids!

File: dashboard/streamlit_app.py [hunk 2]
```diff
@@ -77,6 +81,7 @@ def get_circuit_selection(conn: connect, key: str) -> st.multiselect:
     judge_selection = st.multiselect(key=key,
                                      placeholder="select circuit(s)",
                                      options=rows,
+                                     default=None,
```
Comment (ErvinRex) [inline dashboard/streamlit_app.py:84]: What does this improve?

File: dashboard/streamlit_app.py [hunk 3]
```diff
@@ -139,6 +144,44 @@ def get_judge_type_selection(conn: connect, key: str) -> st.selectbox:
 
     return judge_selection
 
+
+def compile_inputs_as_dict(conn: connect, judge_type: str = None, circuits: str = None,
+                           gender: str = None, date: tuple = None, judge: str = None) -> dict:
+    """Returns input widget returns as a single dictionary"""
+
+    with conn.cursor() as cur:
```
Comment (ErvinRex) [inline dashboard/streamlit_app.py:152]: what happens if this fails? could a try and except prevent this from crashing?

File: dashboard/streamlit_app.py (hunk 2 above)
Comment (RachelPun) [reply dashboard/streamlit_app.py:84]: i think this ensures that the multiselect initialises without any selection, so users would start with an untouched multiselect

File: dashboard/streamlit_app.py [hunk 4]
```diff
@@ -139,6 +144,44 @@ def get_judge_type_selection(conn: connect, key: str) -> st.selectbox:
...
+            cur.execute(query_type, [judge_type])
+            judge_type = cur.fetchone()["judge_type_id"]
+
+        if circuits:
+            placeholders = ', '.join([f"'{circuit}'" for circuit in circuits])
+            st.write(placeholders)
+            query_circuit = f"""
+                            SELECT circuit_id
+                            FROM circuit
+                            WHERE name IN ({placeholders})
```
Comment (ErvinRex) [inline dashboard/streamlit_app.py:168]: Is `IN` better than using `ILIKE` here?

**Review by ErvinRex — APPROVED** [review]
Great developments in the controls, would love to see the catching of errors in case the database goes down.

File: dashboard/streamlit_app.py (hunk 4 above)
Comment (RachelPun) [reply dashboard/streamlit_app.py:168]: I think we have to use `IN` because we are passing in a list to check instead of one single string

**Pull request merged**